src/lib.rs

```diff
@@ -381,18 +370,16 @@ const fn byte2hex(byte: u8, table: &[u8; 16]) -> (u8, u8) {
 /// # }
 /// ```
 pub fn encode_to_slice<T: AsRef<[u8]>>(input: T, output: &mut [u8]) -> Result<(), FromHexError> {
-    if input.as_ref().len() * 2 != output.len() {
+    let input = input.as_ref();
+    if input.len() * 2 != output.len() {
         return Err(FromHexError::InvalidStringLength);
     }
 
-    for (byte, (i, j)) in input
-        .as_ref()
-        .iter()
-        .zip(generate_iter(input.as_ref().len() * 2))
-    {
-        let (high, low) = byte2hex(*byte, HEX_CHARS_LOWER);
-        output[i] = high;
-        output[j] = low;
+    // TODO: use array_chunks_mut i/o chunks_exact_mut once that’s stabilises.
```

Comment:
"i/o" presumably stands for "instead of", but spelling it out makes this much easier to read. Also fix the error at the end of sentence.

Reply:
Done.